tests/mailinglist.rs

```diff
@@ -0,0 +1,332 @@
+#![feature(async_closure)]
+
+extern crate diesel;
+extern crate rocket;
+
+use lettre::message::Mailbox;
+use rocket::http::{Accept, ContentType, Status};
+
+use zauth::config::Config;
+use zauth::mailer::Mailer;
+use zauth::models::mail::NewMail;
+use zauth::models::user::*;
+use zauth::DbConn;
+
+mod common;
+
+const TEST_USERS: [(&'static str, UserState, bool); 10] = [
+	("valid0", UserState::Active, true),
+	("valid1", UserState::Active, true),
+	("pending_approval0", UserState::PendingApproval, true),
+	("pending_approval1", UserState::PendingApproval, true),
+	("pending_email0", UserState::PendingMailConfirmation, true),
+	("pending_email1", UserState::PendingMailConfirmation, true),
+	("disabled0", UserState::Disabled, true),
+	("disabled1", UserState::Disabled, true),
+	("unsubbed0", UserState::Disabled, false),
+	("unsubbed01", UserState::Disabled, false),
```

Comment:
```suggestion
	("unsubbed0", UserState::Active, false),
	("unsubbed01", UserState::Active, false),
```
From what I've gathered, disabled users should not receive emails (right?) so to test the unsubscribe functionality, the unsubbed users should actually be active I think.

Reply:
indeed the should, good catch